src/helpers/primary-term-helper.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Primary term helper.
+ *
+ * @package Yoast\WP\SEO\Helpers
+ */
+
+namespace Yoast\WP\SEO\Helpers;
+
+/**
+ * Class Primary_Term_Helper
+ */
+class Primary_Term_Helper {
+
+	/**
+	 * Returns all the taxonomies for which the primary term selection is enabled.
+	 *
+	 * @param int $post_id Default current post ID.
+	 *
+	 * @return array The taxonomies.
+	 */
+	public function get_primary_term_taxonomies( $post_id = null ) {
+		if ( $post_id === null ) {
+			$post_id = \get_the_ID();
+		}
+
+		return $this->generate_primary_term_taxonomies( $post_id );
+	}
+
+	/**
+	 * Generate the primary term taxonomies.
+	 *
+	 * @param int $post_id ID of the post.
+	 *
+	 * @return array The taxonomies.
+	 */
+	protected function generate_primary_term_taxonomies( $post_id ) {
+		$post_type      = \get_post_type( $post_id );
```

Comment:
If we pass the `indexable` we don't have to look this up.

Reply:
It is not that easy, unfortunately.
The helper is also used in the `primary-term-watcher`, which does not have access to the Indexable. I decided, for now, to leave it as is.